### PR TITLE
Convert (some) test endpoints to use ServeHTTP instead of direct calls to handlers.

### PR DESCRIPTION
**Pull request by Mathew-Estafanous**
Issue #11396

**Overview:**
Initiate the process of converting direct calls to HTTPHandler(s) to calls through ``ServeHTTP(resp, req)`` when testing. There are quite a lot of instances so as requested I just made a handful of changes to keep it small.

**What I Did:**
As mentioned, I converted a few of the tests in ``agent_endpoint_test`` to go through the proper channel. However, as discussed in the [issue comments](https://github.com/hashicorp/consul/issues/11396#issuecomment-953419523) there are many cases where the tests are checking for specific internal objects. As a result I did as recommended and converted the JSON responses back into the object representations to proceed with the tests.

### CODE REVIEW

**Comment by Mathew-Estafanous** [comment]
How do I request a code review for this PR? To my knowledge, I don’t seem to have the ability to do it myself.

**Comment by Mathew-Estafanous** [comment]
Ok @kisunji , I’m ready for a review.

File: agent/agent_endpoint_test.go [hunk 1]
```diff
@@ -5397,13 +5417,9 @@ func TestAgent_Token(t *testing.T) {
...
 
-			_, err := a.srv.AgentToken(resp, req)
-			if tt.expectedErr != "" {
-				require.EqualError(t, err, tt.expectedErr)
-				return
-			}
-			require.NoError(t, err)
+			a.srv.h.ServeHTTP(resp, req)
 			require.Equal(t, tt.code, resp.Code)
+			require.Contains(t, resp.Body.String(), tt.expectedErr)
```
Comment (kisunji) [inline agent/agent_endpoint_test.go:5422]: Could you wrap this in an `if tt.expectedErr != ""` block? We want to compare only if it's not blank and return early (so we don't compare the fields below).

**Review by kisunji — COMMENTED** [review]
Looks good! Just one minor change

File: agent/agent_endpoint_test.go (hunk 1 above)
Comment (Mathew-Estafanous) [reply agent/agent_endpoint_test.go]: On it! I just pushed the change.

**Comment by kisunji** [comment]
Thanks @Mathew-Estafanous! I'll merge it now. Feel free to submit pull requests for other tests (or look around at other issues that interest you)

**Review by kisunji — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/492140.

**Comment by Mathew-Estafanous** [comment]
Thanks so much @kisunji ! I’m considering working on most of the test cases within ``agent_endpoint_test.go`` however I realize that it might turn out to be quite a large change. Would it be ok if I submit all those changes as one PR or would it be more appropriate to split them out into small PRs (similar to this size)?

**Comment by kisunji** [comment]
> Thanks so much @kisunji ! I’m considering working on most of the test cases within `agent_endpoint_test.go` however I realize that it might turn out to be quite a large change. Would it be ok if I submit all those changes as one PR or would it be more appropriate to split them out into small PRs (similar to this size)?

Given your experience now I think one file = one PR is good!

**Pull request merged**